linear_model/Base.go

```diff
@@ -623,7 +623,7 @@ func PreprocessData(X, Y *mat.Dense, FitIntercept, Normalize bool, SampleWeight
 				Xout.Copy(X)
 			}
 			scale := 1.
-			if Normalize {
+			if FitIntercept && Normalize {
```

Comment:
The documentation said Normalization will happen only when the Intercept value is set to true, so I added the additional condition before normalizing the data